### ISSUE TITLE
Slideshow Presentations

### ISSUE THREAD

**Issue by chrisjsewell** [ExecutableBookProject/meta#52]
How/when/what/why

One for the future, but since I'm just writing a presentation now I'll kick this off.

For sphinx, I haven't really come across any good implementations. In terms of Jupyter Notebooks, [RISE](https://rise.readthedocs.io/en/stable/) I think is the most mature implementation, but is obviously very notebook centric.

Stepping back a bit; is there a good/simple Markdown orientated slide show generator? From a quick google, this provides some possible options: https://opensource.com/article/18/5/markdown-slide-generators.

With presentations, the "difficulty" is that you usually want to be able to have more specific control over the layout, which is not as easy with just text-based files. So it would be ideal if it had some kind of two-way relationship with a GUI. Also, obviously the behemoth in the room is Powerpoint, and ideally there would be at least a one-way export to this format.

**Comment by choldgraf** [comment]
Yeah, to be honest I have never found a from-text slides tool that is very satisfying, so I just go back to using Google Slides each time. I am way too picky about making everything look just as I want it to look :-/

But if we want live computation in there and we are fine w/ notebooks as an intermediate then I've found RISE to be pretty nice. It also uses reveal.js under the hood, so it could be that somebody else out there has written a "markdown -> reveal" converter out there

**Comment by rowanc1** [comment]
I was also using reveal for a while, it worked well (ish) with https://slides.com/ which is by the author I believe. There was a good cross over between the GUI and the HTML output. I also found that I needed more control and faster editing for presentations than what I could do through text-editing. But the export feature was quite nice and allowed mixing of interactive graphics and presentation (e.g. https://row1.ca/static/logically-rectangular-mesh/index.html#/1). Very interested in this ...! :)

**Comment by akhmerov** [comment]
Something relevant that changed since the rise of RISE is the appearance of thebelab (+ binder). Of course it doesn't render RISE obsolete, but it opens new possibilities.

**Comment by chrisjsewell** [comment]
Yeh cheers guys, certainly food for thought.

I guess maybe there're two kinds of use cases:

- Creating "live" code presentations, where you certainly need something along the lines of RISE/reveal.js
- Creating "traditional" presentations. Here you're probably never going to beat a Powerpoint or Google Slides, but perhaps there is scope to use standard notebooks and/or our text-based (myst) notebooks to generate "template" presentations in a more one-way export fashion. The Google Slides API looks like an interesting route: https://github.com/gsuitedevs/md2googleslides, and note that this uses `markdown-it` which is now the underlying parser for MyST 😄

**Comment by chrisjsewell** [comment]
>  The Google Slides API looks like an interesting route

Well if I could actually use the damn thing! gsuitedevs/md2googleslides#70

**Comment by choldgraf** [comment]
@chrisjsewell actually I have pursued a "hybrid" approach before which has worked well. I'd have my "main" slides in google slides, and then jump to a RISE presentation for some live demos that worked pretty well

**Comment by cpjobling** [comment]
I'm just starting with the new sphinx Jupyter books and the problem I am going to have is I can't use the MyST features for equations and figures if I want to present certain chapters of a book as slides with embedded executions.

**Comment by choldgraf** [comment]
@cpjobling yep this is going to be a challenge until there is MyST support inside of Jupyter environments (I believe that RISE cells use nbconvert to convert `ipynb` markdown to HTML, and this doesn't speak MyST.

**Comment by choldgraf** [comment]
Just a note that it looks like there is a sphinx revealjs extension here: https://github.com/attakei/sphinx-revealjs

I wonder if we could leverage that to get revealjs books that could use myst markdown under the hood 🤔

**Comment by sina-mansour** [comment]
Hey, I just noticed this issue thread and thought it would be worth pointing to slidev, a fully HTML based presentation framework utilizing markdown:

https://sli.dev/

I can only imagine how cool it could get if the idea of a jupyter book could somehow be also linked to a presentation format such as slidev.
